src/Umbraco.Web/Routing/PublishedContentRequestEngine.cs

```diff
@@ -606,9 +607,8 @@ private void FindTemplate()
 			// only if the published content is the initial once, else the alternate template
 			// does not apply
             // + optionnally, apply the alternate template on internal redirects
-            var useAltTemplate = _webRoutingSection.DisableAlternativeTemplates == false 
-                && (_pcr.IsInitialPublishedContent
-                || (_webRoutingSection.InternalRedirectPreservesTemplate && _pcr.IsInternalRedirectPublishedContent));
+            var useAltTemplate = _pcr.IsInitialPublishedContent

```

Comment:
This one took a bit of head scratching but it is purely to limit duplication and to separate concerns. Assuming an alternative template is actually requested then we will travel to line 650 where `IsTemplateAllowed` is called. `IsTemplateAllowed` is intended to be the process for checking if we can request that template and so it is that that checks `_webRoutingSection.DisableAlternativeTemplates`.